src/matrix/room/BaseRoom.js

```diff
@@ -501,7 +501,9 @@ export class BaseRoom extends EventEmitter {
                 },
                 clock: this._platform.clock,
                 logger: this._platform.logger,
-                powerLevelsObservable: await this.observePowerLevels()
+                powerLevelsObservable: await this.observePowerLevels(),
+                fetchEventFromStorage: eventId => this._readEventById(eventId),
+                fetchEventFromHomeserver: eventId => this._getEventFromHomeserver(eventId)
```

Comment:
I think we can also consider to pass hsApi into timeline for this... unless there's a good reason to inject? Unit tests perhaps?

Reply:
Felt more cohesive to have this code in the same place that we have the code to fetch an event from storage. Would encapsulating these two methods into an `eventFetcher` be more agreeable?

Reply:
So if we end up with readEventById in TimelineReader, I would put this in the Timeline. I don't thing we will ever really need to call /context from outside of the Timeline.